internal/cluster/infrastructure/aws/awsworkflow/activity_calculate_node_pool_version.go

```diff
@@ -12,14 +12,16 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package eksworkflow
+package awsworkflow
```

Comment:
What was the reason behind unifying the EKS and PKE node pool version calculation activities?

As far as I understand those two distributions are being developed separately from each other feature-parity wise which means the node pool operations allow different attributes to be modified and thus the version calculation take different inputs for EKS and PKE.
This means for example EKS node pools support EBS encryption and thus it is a parameter of the version calculation, but PKE/AWS does not yet support this feature and we don't see whether it ever will, so it is always going to be a dummy argument in that context.

I would be fine with having a fully generic hash calculation activity with a string slice input and single string output being shared between distributions or even cloud providerss as it would be distribution- and provider-independent in its interface. (But then I would certainly add distribution-specific wrappers to lock the global input argument list for a single distribution.)
But I warn against coupling distribution specific interfaces together even under the same provider as those components historically tend to develop very much independently of each other and I think dummy values are always unnecessary added complexity and require extra out of band domain knowledge to be handled properly.